### ISSUE TITLE
`_prepare_packed_dataloader` and `_prepare_non_packed_dataloader` have inconsistent keys

### ISSUE THREAD

**Issue by maneandrea** [huggingface/trl#1122]
`_prepare_packed_dataloader`  outputs a `ConstantLengthDataset` with [keys](https://github.com/huggingface/trl/blob/830cadfc4c80bdced0d3753de392070e6760d1f5/trl/trainer/utils.py#L450)
```
'input_ids`, 'labels'
```
`_prepare_non_packed_dataloader` outputs some type of dataset, with [keys](https://github.com/huggingface/trl/blob/830cadfc4c80bdced0d3753de392070e6760d1f5/trl/trainer/sft_trainer.py#L393)
```
'input_ids`, 'attention_mask'
```
The lack of `'labels'` in the latter might be source of confusing bugs.

If this is intended behavior, I would at least suggest to post a warning in the documentation of the argument `packing` of `SFTTrainer`

**Comment by lvwerra** [comment]
I agree it's a bit confusing but it shouldn't make a difference in practice: the `labels` in the first case are identical to `input_ids` which is the default value filled also when no `labels` are provided in the second case. Similarly for the `attention_mask` in the second case. 

If you indeed encountered weird behaviour let us know so we can fix. I think we can refactor this in the future to be a bit more coherent. cc @younesbelkada

**Comment by maneandrea** [comment]
An example of undesired (or, at least, confusing) behavior is the following:
```

def do(packing, collator):
    # some code to prepare the dataset etc....

    trainer = SFTTrainer(
        model=my_model,
        train_dataset=my_dataset,
        eval_dataset=eval_dataset,
        packing=packing,
        dataset_text_field='text',
        data_collator=collator,
    )
    
    trainer.train()
    
do(True, None)                    # works
do(True, DefaultDataCollator())   # works
do(False, None)                   # works
do(False, DefaultDataCollator())  # does not work
```
Even just a warning not to use `DefaultDataCollator` I guess should be enough.

**Comment by lvwerra** [comment]
Indeed, that's not ideal. I'll have a look at it hopefully soon, but if you find time in the meantime feel free to open a PR!

**Comment by maneandrea** [comment]
I did create a PR, see #1174. Let me know if there's anything you need me to do
